code/modules/projectiles/gun_attachables.dm

```diff
@@ -123,6 +123,8 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	var/charge_mod = 0
 	///what firemodes this attachment allows/adds.
 	var/gun_firemode_list_mod = null
+	///lazylist of attachment slot offsets for a gun.
+	var/list/gun_attachment_offset_mod
```

Comment:
isnt this on the attachments already though

Reply:
no, this changes other attachment offsets on the gun